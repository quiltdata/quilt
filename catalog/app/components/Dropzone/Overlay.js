import cx from 'classnames'
import * as React from 'react'
import * as M from '@material-ui/core'

const useStyles = M.makeStyles((t) => ({
  root: {
    alignItems: 'center',
    background: t.palette.action.disabledBackground,
    border: `1px solid ${t.palette.action.disabled}`,
    borderRadius: t.shape.borderRadius,
    bottom: 0,
    cursor: 'not-allowed',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
  },
}))

export default function Overlay({ children, className }) {
  const classes = useStyles()
  return <div className={cx(classes.root, className)}>{children}</div>
}
