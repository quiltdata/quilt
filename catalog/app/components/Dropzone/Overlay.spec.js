import * as React from 'react'
import renderer from 'react-test-renderer'

import Overlay from './Overlay'

describe('components/Dropzone/Overlay', () => {
  it('should render', () => {
    const tree = renderer.create(<Overlay />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should apply className', () => {
    const tree = renderer.create(<Overlay className="alteredStyles" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
