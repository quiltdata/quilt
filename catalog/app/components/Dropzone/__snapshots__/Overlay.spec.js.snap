// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropzone/Overlay should apply className 1`] = `
<div
  className="makeStyles-root-1 alteredStyles"
/>
`;

exports[`components/Dropzone/Overlay should render 1`] = `
<div
  className="makeStyles-root-1"
/>
`;
